storage/storageCacherAdapter/storageCacherAdapter.go

```diff
@@ -0,0 +1,243 @@
+package storageCacherAdapter
+
+import (
+	"math"
+	"sync"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var log = logger.GetOrCreate("storageCacherAdapter")
+
+type storageCacherAdapter struct {
+	cacher storage.AdaptedSizedLRUCache
+	db     storage.Persister
+	lock   sync.RWMutex
+
+	storedDataFactory storage.StoredDataFactory
+	marshalizer       marshal.Marshalizer
+}
+
+// NewStorageCacherAdapter creates a new storageCacherAdapter
+func NewStorageCacherAdapter(
+	cacher storage.AdaptedSizedLRUCache,
+	db storage.Persister,
+	storedDataFactory storage.StoredDataFactory,
+	marshalizer marshal.Marshalizer,
+) (*storageCacherAdapter, error) {
+	if check.IfNil(cacher) {
+		return nil, storage.ErrNilCacher
+	}
+	if check.IfNil(db) {
+		return nil, storage.ErrNilPersister
+	}
+	if check.IfNil(marshalizer) {
+		return nil, storage.ErrNilMarshalizer
+	}
+	if check.IfNil(storedDataFactory) {
+		return nil, storage.ErrNilStoredDataFactory
+	}
+
+	return &storageCacherAdapter{
+		cacher:            cacher,
+		db:                db,
+		lock:              sync.RWMutex{},
+		storedDataFactory: storedDataFactory,
+		marshalizer:       marshalizer,
+	}, nil
+}
+
+// Clear clears the underlying cacher and db
+func (c *storageCacherAdapter) Clear() {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// TODO also clear the underlying db
+	c.cacher.Purge()
+}
+
+// Put adds the given value in the cacher. If the cacher is full, the evicted values will be persisted to the db
+func (c *storageCacherAdapter) Put(key []byte, value interface{}, sizeInBytes int) (evicted bool) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	evictedValues := c.cacher.AddSizedAndReturnEvicted(string(key), value, int64(sizeInBytes))
+	for evictedKey, evictedVal := range evictedValues {
+		evictedKeyStr, ok := evictedKey.(string)
+		if !ok {
+			log.Warn("invalid key type")
```

Comment:
log also the key. so `log.Warn("invalid key type", "key", evictedKey)`

Reply:
Done.